lib/dnsmsg-parser/src/dns_message_parser.rs

```diff
@@ -933,25 +822,35 @@ fn parse_dns_update_message_header(dns_message: &TrustDnsMessage) -> UpdateHeade
     }
 }
 
-fn parse_edns(dns_message: &TrustDnsMessage) -> Option<OptPseudoSection> {
-    dns_message.edns().map(|edns| OptPseudoSection {
-        extended_rcode: edns.rcode_high(),
-        version: edns.version(),
-        dnssec_ok: edns.dnssec_ok(),
-        udp_max_payload_size: edns.max_payload(),
-        options: parse_edns_options(edns),
-    })
+fn parse_edns(dns_message: &TrustDnsMessage) -> DnsParserResult<Option<OptPseudoSection>> {
+    dns_message
+        .edns()
+        .map(|edns| {
+            parse_edns_options(edns).map(|options| OptPseudoSection {
+                extended_rcode: edns.rcode_high(),
+                version: edns.version(),
+                dnssec_ok: edns.dnssec_ok(),
+                udp_max_payload_size: edns.max_payload(),
+                options,
+            })
+        })
+        .transpose()
 }
 
-fn parse_edns_options(edns: &Edns) -> Vec<EdnsOptionEntry> {
+fn parse_edns_options(edns: &Edns) -> DnsParserResult<Vec<EdnsOptionEntry>> {
     edns.options()
         .as_ref()
         .iter()
         .map(|(code, option)| match option {
             EdnsOption::DAU(algorithms)
             | EdnsOption::DHU(algorithms)
-            | EdnsOption::N3U(algorithms) => parse_edns_opt_dnssec_algorithms(*code, *algorithms),
-            EdnsOption::Unknown(_, opt_data) => parse_edns_opt(*code, opt_data),
+            | EdnsOption::N3U(algorithms) => {
+                Ok(parse_edns_opt_dnssec_algorithms(*code, *algorithms))
+            }
+            EdnsOption::Unknown(_, opt_data) => Ok(parse_edns_opt(*code, opt_data)),
+            option => Err(DnsMessageParserError::UnknownEdnsOption {
+                option: option.clone(),
+            }),
```

Comment:
They made this enum non-exhaustive. If they do add another member, then we would need to handle it. Rust is adding a lint for this, but it isn't stable yet: https://github.com/rust-lang/rust/issues/89554

Reply:
Actually, I was able to handle the unknown options generically in [9ec889d](https://github.com/vectordotdev/vector/pull/11604/commits/9ec889db61f9c780fece75ee5d572ce31410eea2) which improves this. Ideally, we'd still get compiler warnings if they add a new variant, but this at least removes the error I was previously raising up.